### PR TITLE
Add the ability to filter by all available fields of a model with mod…

### PR DESCRIPTION
**Pull request by pctSW1**
Add the ability to filter by all available fields of a model with modelviewset and djangorestframework `filter_fields='__all__'`

### CODE REVIEW

**Comment by pctSW1** [comment]
I have no idea why this failed....

File: url_filter/integrations/drf.py [hunk 1]
```diff
@@ -87,6 +87,23 @@ def get_filter_class(self, view, queryset=None):
         if filter_class:
             return filter_class
 
+        if filter_fields == '__all__':
```
Comment (miki725) [inline url_filter/integrations/drf.py]: thanks for the contribution!

the intent here is great however it seems to be duplicating the if statement below very closely. easier change might be to add support in `ModelFilterSet` to support `__all__` as a valid value. from the quick look only required change will be in

https://github.com/miki725/django-url-filter/blob/ba476094adeb0db4bb39ee0f4c3d49681d1bd1ba/url_filter/filtersets/base.py#L470

to support both `None` and `__all__`:

```suggestion
if self.Meta.fields in [None, '__all__']:
```

let me know if you will need any help with tests or anything else.

thanks again

Comment (pctSW1) [reply url_filter/integrations/drf.py]: your quite welcome!  

it is duplicating the if statement pretty closely, I tried to pass None in for my ModelViewSet's filter_fields and I was not able to filter on any of the columns.  Thus why I changed what I changed.  The below didn't work (please let me know if I'm using this correctly)
```
class ParamGroupScpiCommandsViewSet(ModelViewSet):
    serializer_class = ParamGroupScpiCommandsSerializer
    queryset = ParamGroup_ScpiCommands.objects.all()
    filter_backends = [DjangoFilterBackend]
    filter_fields = None

```
but after the code edit that I submitted 

```
class ParamGroupScpiCommandsViewSet(ModelViewSet):
    serializer_class = ParamGroupScpiCommandsSerializer
    queryset = ParamGroup_ScpiCommands.objects.all()
    filter_backends = [DjangoFilterBackend]
    filter_fields = '__all__'
```

works

I left the original code in tact as to not corrupt anyone's existing filters that they have set up.  

Side note, I'm totally going to need help writing tests lol

Comment (pctSW1) [reply url_filter/integrations/drf.py]: I just edited this area in the code, the interesting part here is when the filterset gets created using None the primarykey field (Django id column) doesn't get added to the filterable fields.  This is why I thought my solution worked correctly.  After changing using your logic, the id field is still non existent.  I love this library and have just been trying to get around to helping out where I can so thank you for your support and great library!

Comment (pctSW1) [reply url_filter/integrations/drf.py]: I checked in another change to the same branch in my repository but I don't know why its not showing that new commit

Comment (miki725) [reply url_filter/integrations/drf.py]: > I love this library and have just been trying to get around to helping out where I can so thank you for your support and great library!

awesome. glad you enjoy it 😄 

and thank *you* for helping out. any contribution is welcome.

**Comment by pctSW1** [comment]
@miki725 I do have a question for you, when I utilize the filter_fields = None currently I cannot filter on the id field that django creates.  Is this because you can just reach that object by using `/<id>/` ?  I also am kinda confused about creating a test case for this.  I checked in my code and added myself to the authors, please let me know if there is anything I'm missing and as to why I cannot or should not have a filter where I can query based on /?id=***.  Thank you!

**Comment by pctSW1** [comment]
@miki725  Also I'm finding that when I pass None in for the filter fields attribute then I have nothing to filter on because when None is passed into filter fields then this piece of code never gets called 

      ```
  if filter_fields:
            model = filter_class_default.filter_backend_class(queryset).get_model()

            meta_kwargs = filter_class_meta_kwargs.copy()
            meta_kwargs.update({
                'model': model,
                'fields': filter_fields,
            })
            meta = type(str('Meta'), (object,), meta_kwargs)

            return type(
                str('{}FilterSet'.format(model.__name__)),
                (filter_class_default,),
                {'Meta': meta}
            )
```

Please advise.

**Comment by miki725** [comment]
in DRF viewset you will want to pass `'__all__'`, not `None`. that will create all filters which should include `id` field which you can see in your test case.

not sure if you are passing `__all__` anywhere else

**Comment by pctSW1** [comment]
Absolutely, I just thought I could pass either None or '__all__' based on this comment [Comment On Enhancement](https://github.com/miki725/django-url-filter/issues/39#issuecomment-310389406).  But your right after updating the code to accept all it works as expected just not using None but using the __all__ logic.  What confuses me is that  
```if self.Meta.fields in [None, '__all__']:
            self.Meta.fields = self._get_model_field_names()
```
because self.Meta.fields would could never be null because of the previous comment code example

**Comment by miki725** [comment]
you can pass `None` or `__all__` only when creating filterset directly. in DRF only `__all__` will work

**Comment by pctSW1** [comment]
Thank you so much for your help @miki725 I hope if I can, help out more.  I'm going to look into the date issue too here shortly and try to make another pull request to fix that after you merge this one in.

**Pull request merged**